draft-ietf-webpush-protocol.xml

```diff
@@ -556,6 +556,95 @@ TTL = 1*DIGIT
+          push message with a topic replaces any outstanding push message with
+          an identical topic.
+        </t>
+        <t>
+          A push message topic is a string.  A topic is carried in a Topic
+          header field.  A topic is used to correlate push messages sent to the
+          same subscription, it does not convey any other semantics.
+        </t>
+        <t>
+          The grammar for the Topic header field uses the <spanx
+          style="verb">token</spanx> and <spanx
+          style="verb">quoted-string</spanx> rules defined in <xref
+          target="RFC7230"/>.
+        </t>
+        <figure>
+          <artwork type="abnf">
+            <![CDATA[
+Topic = token / quoted-string
+          ]]>
+          </artwork>
+        </figure>
+        <t>
+          Any double quotes from the <spanx style="verb">quoted-string</spanx>
+          form are removed before comparing topics for equality.
+        </t>
+        <t>
+          A push message request creates a new push message resource, but
+          simultaneously deletes any existing message resource that has a
+          matching topic.  In effect, the information that is stored for the
+          push message is updated, but a new resource is created to avoid
+          problems with in flight acknowledgments for the old message.  The push
+          service MAY suppress acknowledgement receipts for the replaced
+          message.
+        </t>
+        <t>
+          This mechanism doesn't allow for stored push messages to be deleted.
+          An application server can send a DELETE request toward a push message
+          URL to remove (or acknowledge) an undelivered message.  Note that a
+          push message with a topic that has a zero or absent time-to-live
+          causes any stored message with the same topic to be removed, though a
+          push message with zero time-to-live could be forwarded to the user
+          agent.
+        </t>
+        <t>
+          A push message with a topic that is not shared by an outstanding
+          message to the same subscription is stored or delivered as normal.
+        </t>
+        <figure>
+          <preamble>
+            For example, the following message could cause an existing message
+            to be updated:
+          </preamble>
+          <artwork type="inline"><![CDATA[
+POST /p/JzLQ3raZJfFBR0aqvOMsLrt54w4rJUsV HTTP/1.1
+Host: push.example.net
+TTL: 600
+Topic: "upd"
+Content-Type: text/plain;charset=utf8
+Content-Length: 36
+
+ZuHSZPKa2b1jtOKLGpWrcrn8cNqt0iVQyroF
+]]></artwork>
+        </figure>
+        <figure>
+          <preamble>
+            If the push service identifies an outstanding push message with a
+            topic of "upd", the contents of that message are replaced.  The push
+            service then responds with a 201 status code, identifying the push
+            message resource that was created in a Location header field.
+          </preamble>

```

Comment:
s/The push service then responds/In all cases, the push service then responds/
To make clear that even if an outstanding push message is replaced, the push service still responds with the 201 status code.